scrapy/commands/shell.py

```diff
@@ -13,6 +15,30 @@
 from scrapy.utils.spider import spidercls_for_request, DefaultSpider
 
 
+def guess_scheme(url):
```

Comment:
What do you think about moving this function to scrapy.utils.url? It may be convenient to use this function to process spider arguments or arguments of custom commands.